source/sync/migrate-sync-modes.txt

```diff
@@ -0,0 +1,356 @@
+.. _realm-sync-migrate-modes:
+
+=========================
+Migrate Device Sync Modes
+=========================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 3
+   :class: singlecol
+
+Atlas App Services Apps can have one of two Device Sync modes: Partition-Based Sync
+and Flexible Sync. Partition-Based Sync is an older mode and you should consider
+migrating to Flexible Sync.
+
+Migrating an App Services App that uses Partition-Based Sync to Flexible Sync is
+an automatic process. Other than upgrading the SDK version, migrating Sync modes
+doesn't require any changes to your client app code.
+
+Migration results in Partition-Based Sync clients communicating with a
+Flexible Sync backend. We recommend eventually :ref:`migrating your client app 
+<realm-sync-migrate-client>` to also use Flexible Sync instead of
+Partition-Based Sync.
+
+Requirements
+------------
+
+Currently, the Kotlin and Swift SDKs can migrate from Partition-Based Sync to
+Flexible Sync. The C++, Flutter, Java, .NET, Node.js, and React Native SDKs
+cannot perform the migration.
+
+App Services App:
+
+- Currently use Partition-Based Sync
+
+Linked Atlas Cluster:
+
+- Use MongoDB 5.0 or later for Flexible Sync compatibility
+- Have storage auto-scaling enabled
+- Be an M10+ Cluster Tier
+- Have time-based oplog setting enabled
+
+Client apps:
+
+- ``realm-core`` v13.10.0 or later, which is included in these SDK versions:
+
+  - Realm Kotlin SDK v1.8.0
+  - Realm Swift SDK v10.39.0
+
+.. _realm-sync-migrate-what-to-know:
+
+What to Know Before Migrating
+-----------------------------
+
+Migrating your App's Sync mode involves changes that affect both client apps and
+your App's backend. You should be aware of the effects and plan ahead.
+
+- Client apps that use an SDK with with a ``realm-core`` version earlier than
+  v13.10.0 will not be able to sync after migration.
+- How long a migration takes is directly proportional to the amount of data that
+  needs to be migrated. The more data, the longer it will take to migrate.
+- Migrations can be :ref:`canceled <realm-sync-migrate-cancel>` during the 
+  migration or :ref:`reverted <realm-sync-migrate-revert>` within a certain 
+  timeframe after the migration is finished.
+- Progress notifications will not work after migration because Flexible Sync does
```

Comment:
@kmorkos, would having client code for progress notifications cause fatal client errors? I know one of the goals of the project was to prevent devs from needing to make client code changes. Is this something that has to be done?

Reply:
It won't trigger any errors, the handlers will just always report 100% progress. So they can keep it in, it will just be unreliable

Reply:
I think I'll leave this where it is, then. It's not great to have unnecessary client code, but removing it isn't exactly required.